gen/cycle.ml

```diff
@@ -463,13 +463,20 @@ module CoSt = struct
 
   let get_cell st = st.co_cell
 
-  let set_cell st e = match e.bank with
-    | Ord ->
-       let old = st.co_cell.(0) in
-       let cell = E.overwrite_value old e.atom e.v in
-       let co_cell = Array.copy st.co_cell in
-       co_cell.(0) <- cell ;
-       {e with cell=co_cell;},{ st with co_cell; }
+  let set_cell ?(rv = -1) st n =
+    let e = n.evt in  match e.bank with
```

Comment:
Just to keep this diff to a minimum, maybe we could move this let further down where it's used. Also many of the lines below we're only changing their indentation.